ember-resources/src/core/class-based/resource.ts

```diff
@@ -195,6 +197,27 @@ export class Resource<Args = unknown> {
     thunk: AsThunk<ArgsFrom<SomeResource>>
   ): SomeResource;
 
+  /**
+   * For use in the body of a class, without reactive arguments.
+   *
+   * `from` is what allows resources to be used in JS, they hide the reactivity APIs
+   * from the consumer so that the surface API is smaller.
+   *
+   * ```js
+   * import { Resource } from 'ember-resources';
+   *
+   * class SomeResource extends Resource {}
+   *
+   * class MyClass {
+   *   data = SomeResource.from(this);
+   * }
+   * ```
+   */
+  static from<SomeResource extends Resource<any>, _C extends Context>(
+    this: Constructor<SomeResource>,
+    contextOrThunk: _C extends Thunk ? AsThunk<ArgsFrom<SomeResource>> : _C
+  ): SomeResource;
```

Comment:
As a result for `Context` being very broadly typed due to `destroyable` (linked above), this required a second generic parameter here to test against. Basically, the same condition used in the code:

https://github.com/NullVoxPopuli/ember-resources/blob/6fc67cb91466ffd1bfcaa2fdee594fdaf7c9dcfc/ember-resources/src/core/class-based/resource.ts#L268

but here on type-level.

Reply:
yeah, function vs object typings are hard :(